lib/pbench/server/auth/__init__.py

```diff
@@ -375,5 +375,6 @@ def token_introspect(self, token: str) -> JSON:
             jwt.ExpiredSignatureError,
             jwt.InvalidSignatureError,
             jwt.InvalidAudienceError,
+            jwt.InvalidAlgorithmError,
         ) as exc:
             raise OpenIDTokenInvalid() from exc
```

Comment:
At the moment, these are the only 4 I can think of that should be a candidate for going to our internal API key decode function.

Reply:
The problem is, the way the code is currently constructed, any error not listed here will cause the server to reject the token without checking to see if it is an API Key.  That strikes me as kind of fragile.

Unfortunately, I'm not aware of any way to separate the "this is not a valid token" exceptions from the "OMG something bad happened!" exceptions, and so you should consider reworking `verify_auth_oidc()` as well , so that it is less sensitive to the result of `token_introspect()`.

Reply:
It's an interesting point which I almost mentioned earlier. Would it be so terrible to try the API key decryption when the OIDC decryption fails *for any reason*?

Reply:
In the absence of some way to detect "OMG WTF BBQ" errors in JWT, I concur with Dave that simply blundering forwards to see if it's "the other kind" is entirely appropriate.

(It would be better still if we could tell which kind of token it was without having to attempt (and fail) a decode operation...for instance, we could prefix API keys with a few characters which we could recognize (and remove before decoding), if we could come up with a string which wouldn't appear there normally.)

Reply:
Yeah, that's a good point I think it's not bad to try to decrypt the API_KEY every time OIDC decryption fails.

Reply:
And...what are your thoughts about doing that in _this_ PR?  😁

Reply:
Maybe not in this PR, but from our earlier discussion, I think I need to investigate how we can move away from the httpAuth package. I need to get this one in so I don't block Dave in some of his functional tests.

Reply:
See #3402.